extensions/skyportal/skyportal/handlers/api/kowalski_filter.py

```diff
@@ -50,46 +59,42 @@ def get(self, filter_id):
                 application/json:
                   schema: Error
         """
-        if filter_id is not None:
-            f = (
-                DBSession()
-                .query(Filter)
-                .filter(
-                    Filter.id == filter_id,
-                    Filter.group_id.in_(
-                        [g.id for g in self.current_user.accessible_groups]
-                    ),
-                )
-                .first()
-            )
-            if f is None:
-                return self.error("Invalid filter ID.")
-
-            group_id = f.group_id
-
-            base_url = (
-                f"{self.cfg['app.kowalski.protocol']}://"
-                f"{self.cfg['app.kowalski.host']}:{self.cfg['app.kowalski.port']}"
+        f = (
+            DBSession()
+            .query(Filter)
+            .filter(
+                Filter.id == int(filter_id),
+                Filter.group_id.in_(
+                    [g.id for g in self.current_user.accessible_groups]
+                ),
             )
-            headers = {"Authorization": f"Bearer {self.cfg['app.kowalski.token']}"}
-
-            url = os.path.join(base_url, f"api/filters/{group_id}/{filter_id}")
-
-            resp = s.get(url, headers=headers, timeout=5)
-
-            if resp.status_code == requests.codes.ok:
-                data = bj.loads(resp.text).get("data")
-                return self.success(data=data)
-            else:
-                message = bj.loads(resp.text).get("message")
-                log(f"Failed to fetch data from Kowalski: {message}")
-                return self.error(f"Failed to fetch data from Kowalski: {message}")
+            .first()
+        )
+        if f is None:
+            return self.error("Invalid filter ID.")
+        response = kowalski.api(
+            method="get",
+            endpoint=f"api/filters/{filter_id}",
+        )
+        data = response.get("data")
+        # drop monogdb's _id's which are not (default) JSON-serializable
+        if data is not None:
+            data.pop("_id", None)
+        status = response.get("status")
+        message = response.get("message")
+
+        if status == "success":
+            return self.success(data=data)
+        return self.error(message=message)
```

Comment:
```suggestion
        if status == "error":
            message = response.get("message")
            return self.error(message=message)
        return self.success(data=data)
```
A bit clearer, and this avoids assigning `message` when it's never accessed in the `success` case

Reply:
Fixed